clients/src/main/java/org/apache/kafka/common/utils/ByteBufferOutputStream.java

```diff
@@ -105,7 +105,7 @@ public int initialCapacity() {
         return initialCapacity;
     }
 
-    private void maybeExpandBuffer(int remainingRequired) {
+    public void maybeExpandBuffer(int remainingRequired) {
```

Comment:
I think it is a bad idea to let callers concern about calling `maybeExpandBuffer`; previously the caller is abstracted from this away.